File: 2946_Matrix_Similarity_After_Cyclic_Shifts/solution.c
bool areSimilar(int** mat, int matSize, int* matColSize, int k) {
    int n = matColSize[0];
    k %= n;
    for (int i = 0; i < matSize; i++) for (int j = 0; j < n; j++) {
        if (mat[i][j] != mat[i][(j + k) % n]) return false;
    }
    return true;
}

File: 2946_Matrix_Similarity_After_Cyclic_Shifts/solution.py
class Solution:
    def areSimilar(self, mat: List[List[int]], k: int) -> bool:
        n, k = len(mat[0]), k % len(mat[0])
        return all(mat[i][j] == mat[i][(j + k) % n if i % 2 == 0 else (j - k) % n] for i in range(len(mat)) for j in range(n))